package/network/services/dnsmasq/files/dnsmasq.init

```diff
@@ -787,6 +787,28 @@ dhcp_hostrecord_add() {
 	xappend "--host-record=$record"
 }
 
+dhcp_dnsrr_add() {
+	#This adds arbitrary resource record types (of IN class) whose optional data must be hex
+	local cfg="$1"
+	local dnsrr rrnumber hexdata
+
+	config_get dnsrr "$cfg" dnsrr
+	[ -n "$dnsrr" ] || return 0
+
+	config_get rrnumber "$cfg" rrnumber
+	[ -n "$rrnumber" ] && [ "$rrnumber" -gt 0 ] || return 0
+
+	config_get hexdata "$cfg" hexdata
+
+	# dnsmasq accepts XX:XX, XX XX, or XXXX forms or mixtures thereof
+	if [ -n "${hexdata//[0-9a-fA-F\:\ ]/}" ]; then
```

Comment:
I would use a `case` and exact globs matching those patterns.

Reply:
Specifically:

```
case $hexdata in
[0-9a-fA-F][0-9a-fA-F][ :][0-9a-fA-F][0-9a-fA-F]|[0-9a-fA-F][0-9a-fA-F][0-9a-fA-F][0-9a-fA-F])
    # handle as valid data
    ;;
*)
   # handle as an error
esac
```

Reply:
Ah, sort of how I imagined. Thank you. 

Then, I believe it is OK as is. hexdata can be much longer strings, just that each byte can be space or colon separated, or not at all. dnsmasq takes all. 

One could just accept a single format to simplify.